### PR TITLE
Adopt API-156: singleton resources

### LINKED ISSUE #137: Adopt AIP-0156 Singleton resources
AEPs [121](https://aep.dev/121) and [122](https://aep.dev/122) refer to "singleton resources", but the definition is missing.

https://github.com/aep-dev/aep.dev/blob/6b3dc9abcc7680d67c9384540e37589321b52915/aep/general/0121/aep.md.j2#L71
https://github.com/aep-dev/aep.dev/blob/6b3dc9abcc7680d67c9384540e37589321b52915/aep/general/0121/aep.md.j2#L149
https://github.com/aep-dev/aep.dev/blob/6b3dc9abcc7680d67c9384540e37589321b52915/aep/general/0122/aep.md.j2#L22
https://github.com/aep-dev/aep.dev/blob/6b3dc9abcc7680d67c9384540e37589321b52915/aep/general/0122/aep.md.j2#L49

### PR DESCRIPTION
**Pull request by HenryGessau**
Closes #137 

This PR has deliberate commits:

1.  Copy AIP-156 from google.aip.dev 
2.  Rename .md file to .md.j2 
3.  AEP-ify 
4.  Add oas examples 

This is my first contribution to this repo. Please be gentle. 😄

### CODE REVIEW

File: aep/general/0156/aep.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+id: 156
+state: reviewing
```
Comment (rofrankel) [inline aep/general/0156/aep.yaml:3]:
```suggestion
state: approved
```

I think we decided not to really use the reviewing state (at least for adopting existing guidance with trivial changes) as a second layer after reviewing a PR.

File: aep/general/0156/aep.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+slug: resource-design
```
Comment (rofrankel) [inline aep/general/0156/aep.yaml:4]:
```suggestion
slug: singletons
```

Slugs are intended to replace numbers as the primary way of referring to AIPs (still a WIP).  They also turn into a short link; this one will be aep.dev/singletons.

File: aep/general/0156/aep.md.j2 [hunk 3]
```diff
@@ -94,6 +144,35 @@ message ListConfigsRequest {
   // other standard pagination fields...
```
Comment (rofrankel) [inline aep/general/0156/aep.md.j2:144]: They're both outside the window where GitHub will let me add a suggestion, but could you update both instances of "api.googleapis.com" to "api.example.com"?

**Review by rofrankel — APPROVED** [review]
Thanks, a few minor comments, but in general this looks great.

I believe @toumorokoshi doesn't love the singleton pattern, but I think it's useful to have, and an organization can always decide to ban the pattern if they don't like it.

File: aep/general/0156/aep.md.j2 (hunk 3 above)
Comment (HenryGessau) [reply aep/general/0156/aep.md.j2:144]: Done.

**Comment by toumorokoshi** [comment]
> Thanks, a few minor comments, but in general this looks great.
> 
> I believe @toumorokoshi doesn't love the singleton pattern, but I think it's useful to have, and an organization can always decide to ban the pattern if they don't like it.

yep, I'd like to revisit at some point, but I don't want that to get in the way of this PR - we've agreed to leave disagreements to issue and adopt AIP guidance as-is as the default.

@rofrankel go ahead and merge. 

Thanks for the contribution @HenryGessau!

**Pull request merged**